### ISSUE TITLE
Natural Armor needs a flag.

### ISSUE THREAD

**Issue by augustjohnson** [open5e/open5e-api#600]
Creatures sometimes have "natural armor" as a tag next to their armor class.  This should be a flag in the API.
This work includes added the field and the data.

**Comment by calumbell** [comment]
This is a good idea, but I think it might be better to use a string insead of a boolean flag. We could then use the field to handle all of the Armor Class suffixes that appear in the dataset in a regular way.

A `"natural_armor"` flag might work if the only AC suffix was *(natural armor)*:

(Aboleth statblock WotC SRD 5.1 pp. 261)

<img width="412" alt="Image" src="https://github.com/user-attachments/assets/78fca35a-3e7e-48e8-bbe2-109bd49e6f6c" />

But then how would we handle a statblock like the SRD Lizardfolk?

(Lizardfolk statblock, WOTC SRD 5.1 pp. 326)

<img width="415" alt="Image" src="https://github.com/user-attachments/assets/b84c225b-4cb6-400e-8a8b-22c1a4bea36d" />

Would we include an additional `"has_shield"` flag? Then how would we handle the Veteran statblock:


(Veteran statblock, WOTC SRD 5.1 pp. 403)
<img width="431" alt="Image" src="https://github.com/user-attachments/assets/d28ba21e-1100-4ffa-9d52-d2740f055aca" />

If each of the possible suffixes has its own flag, the API response will become very bloated.

I think a `"armor_detail"` text field where the armor class suffix is included as a string would be a preferable options. Just my 2c

**Comment by augustjohnson** [comment]
Agreed, the approach should definitely be armor_detail or something along those lines.
